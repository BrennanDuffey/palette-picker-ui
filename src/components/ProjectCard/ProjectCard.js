import React, { Component } from 'react';
import { connect } from 'react-redux';
import { deletePalette } from '../../thunks/deletePalette';
import PalettePreview from '../PalettePreview/PalettePreview';

export class ProjectCard extends Component {
  handleDeletePalette = (paletteId) => {
    this.props.deletePalette(paletteId)
  }

  render() {
    let palettesDisplay;
    const { palettes } = this.props;

    if (palettes.length) {
      palettesDisplay = palettes.map(palette => 
        <PalettePreview 
          handleDeletePalette={this.handleDeletePalette}
          colors={palette.colors_array} 
          paletteId={palette.id}
          key={palette.id} 
        />
      );
    } else {
      palettesDisplay = 
        <h4 className="empty-project-header">
          This project has no saved palettes!
        </h4>
    }

    return (
      <article className="ProjectCard">
        <h3>
          {this.props.project.name}
        </h3>
        <hr />
        <div className="previews-container">
         {palettesDisplay}
        </div>
      </article>
    );
  }
}

export const mapDispatchToProps = (dispatch) => ({
  deletePalette: (paletteId) => dispatch(deletePalette(paletteId))
})

export default connect(undefined, mapDispatchToProps)(ProjectCard);