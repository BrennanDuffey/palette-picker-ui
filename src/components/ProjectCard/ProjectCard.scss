.ProjectCard {
  background: white;
  padding: 50px;

  h3 {
    text-transform: uppercase;
  }

  hr {
    border: solid $lightGray 1px;
    margin: 10px 0;
  }

  .empty-project-header {
    font-style: italic;
    margin: 20px;
    text-align: center;
  }
}
