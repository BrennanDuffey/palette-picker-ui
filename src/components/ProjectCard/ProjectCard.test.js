import React from 'react';
import { shallow } from 'enzyme';
import { ProjectCard } from './ProjectCard';

describe('ProjectCard', () => {
  let mockProject;
  let mockPalettes;
  let wrapper;

  beforeEach(() => {
    mockProject = [
      { 
        id: 1,
        name: 'Project 1', 
      }
    ];
    mockPalettes = [
      {
        id: 11,
        name: 'Palette 1',
        colors_array: [
          '#FDC58D', 
          '#FFFFAA', 
          '#95E38A', 
          '#779DF2', 
          '#D2BEFD'
        ],
        project_id: 1
      }
    ];
    wrapper = shallow(
      <ProjectCard 
        project={mockProject}
        palettes={mockPalettes}
      />
    );
  });

  it('should match snapshot if there are palettes', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should match snapshot if there are no palettes', () => {
    wrapper = shallow(
      <ProjectCard
        project={mockProject}
        palettes={[]}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });
});