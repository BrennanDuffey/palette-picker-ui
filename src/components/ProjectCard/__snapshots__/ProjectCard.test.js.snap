// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectCard should match snapshot 1`] = `
<article
  className="ProjectCard"
>
  <h3>
    Project Name
  </h3>
  <hr />
  <h4
    className="empty-project-header"
  >
    This project has no saved palettes!
  </h4>
  <div
    className="previews-container"
  >
    <PalettePreview />
    <PalettePreview />
    <PalettePreview />
    <PalettePreview />
  </div>
</article>
`;
